src/OSPSuite.Presentation/Presenters/Importer/ImporterPresenter.cs

```diff
@@ -216,18 +216,22 @@ private void loadSheets(IDataSourceFile dataSourceFile, Cache<string, DataSheet>
             return;
          }
 
-         var mappings = dataSourceFile.Format.Parameters.OfType<MetaDataFormatParameter>().Where(p => p.ColumnName != null).Select(md => new MetaDataMappingConverter()
-         {
-            Id = md.MetaDataId,
-            Index = sheetName => md.IsColumn ? dataSourceFile.DataSheets[sheetName].RawData.GetColumnDescription(md.ColumnName).Index : -1
-         }).Union
+         var dataMappings = dataSourceFile.Format.Parameters.OfType<MetaDataFormatParameter>().Where(p => p.ColumnName != null).Select(md =>
+            new MetaDataMappingConverter()
+            {
+               Id = md.MetaDataId,
+               Index = sheetName => md.IsColumn ? dataSourceFile.DataSheets[sheetName].RawData.GetColumnDescription(md.ColumnName).Index : -1
+            }).ToList();
+            
+         var mappings   = dataMappings.Union
          (
             dataSourceFile.Format.Parameters.OfType<GroupByDataFormatParameter>().Select(md => new MetaDataMappingConverter()
             {
-               Id = md.ColumnName,
+               Id = dataMappings.Any(x=> x.Id == md.ColumnName) ? md.ColumnName + Constants.ImporterConstants.GroupingBySuffix : md.ColumnName,
```

Comment:
We have extensions methods such as ExistsById etc... 
I think a comment explaining WHY you are doing the concat here makes sense (*e..g because of mapping to a column already mapped etc)

Reply:
MetaDataMappingConverter was not implementing IWithId (although it did have an Id), that's why I could not find the method in autocomplete. This is a good tip, I need to keep learning things like this that we have already implemented :) .
Also comment added